momento/src/commands/cloud_linter/linter_cli.rs

```diff
@@ -221,12 +249,57 @@ async fn process_data(
         Arc::clone(&control_plane_limiter),
         Arc::clone(&metrics_limiter),
         sender.clone(),
+        metrics_start_millis,
+        metrics_end_millis,
     )
     .await?;
 
     Ok(())
 }
 
+/// Calculates a metric time range based on optional start and end dates.
+/// If start_date is not provided, it defaults to end_date - 30 days.
+/// If end_date is not provided, it defaults to now.
+///
+/// # Arguments
+///
+/// * `start_date` - An optional String representing the start date in "YYYY-MM-DD" format.
+/// * `end_date` - An optional String representing the end date in "YYYY-MM-DD" format.
+///
+/// # Returns
+///
+/// A Result containing a tuple of the start and end timestamps in millis, or a CliError
+/// if date parsing fails.
+fn get_metric_time_range(
+    start_date: Option<String>,
+    end_date: Option<String>,
+) -> Result<(i64, i64), CliError> {
+    let now = Utc::now().naive_utc();
+    let thirty_days = chrono::Duration::days(30).num_milliseconds();
+
+    let processed_end_date = end_date
+        .map(|date| parse_date_string(&date))
+        .transpose()?
+        .unwrap_or(now)
+        .timestamp_millis();
+    let processed_start_date = start_date
+        .map(|date| parse_date_string(&date))
+        .transpose()?
+        .map(|date| date.timestamp_millis())
+        .unwrap_or_else(|| processed_end_date - thirty_days);
+
+    Ok((processed_start_date, processed_end_date))
+}
+
+fn parse_date_string(date: &str) -> Result<NaiveDateTime, CliError> {
+    let naive_date = NaiveDate::parse_from_str(date, "%Y-%m-%d").map_err(|_| CliError {
+        msg: "Date must be in ISO 8601 (YYYY-MM-DD) format".to_string(),
```

Comment:
```suggestion
        msg: "Date must be in YYYY-MM-DD format".to_string(),
```

The `YYYY-MM-DD` is more commonplace than the ISO specification number hence more user friendly.